CIP-0037/CIP-0037.md

```diff
@@ -0,0 +1,338 @@
+
+## Rogue-key attack resistant BLS
+To prevent the rogue-key attack, one can use PoP of the secret keys. However, there are other mechanisms, as 
+described [here](https://eprint.iacr.org/2018/483.pdf). For this, verification is as above, with the difference 
+that key aggregation requires multiscalar multiplications of size n, in both G_1 and G_2, for n the number of 
+submitted signatures.
+
+## ATMS
+ATMS is a k-of-t threshold signature scheme (meaning that we need at least k signers to participate). Let 
+n <= t - k be the number of signers that did not participate in the ATMS signature. ATMS verification proceeds 
+as follows: 
+* Check that n <= t - k 
+* n (log_2(t) hash computations + equality checks) (this is a very pessimistic upper bound. In practice it's much less)
+* n G_1 additions and 1 G_1 negation
+* 2 pairing evaluations + equality check
+
+## Mithril (using trivial proving system)
+Let k be the number of submitted signatures (as above, k is a security parameter determining the number of 
+required signatures). However, in mithril k << N where N is the total number of eligible signers. Mithril 
+verification goes as foollows: 
+* Check the k is large enough
+* k G_1 additions
+* k G_2 additions
+* k (log_2(N) hash computations + equality checks)
+* 2 pairing evaluations + equality check
+
+There are some additional t checks of lottery winning (similar to VRF). These are non crytpgraphic operations 
+(rather inequality checks of indices, and checks that some hash computation is below a threshold).
+
+## Coconut certs
+We use this as an example for anonymous credentials. There is no decision on what type of construction we will 
+eventually use. Coconut credentials (and other types of credentials we are looking at) are built in a way that 
+it is efficient to prove statements about attributes. To this end, it is required to build, and be able to verify, 
+relations over discrete logarithm values. However, for sake of simplicity in this computation, we consider the 
+simplest form of anonymous credentials, which contains no attributes. We not that proving statements regarding 
+attributes does not require further pairing evaluations. 
+* Proof verification to a relation involving elements in G_1 and G_2. This verification does not require the 
+* pairing evaluation, but does require G_1 and G_2 additions and multiplications. 
+* 2 pairing evaluations + equality check
+
+The complexity of the statement one can prove with anonymous credentials will be bounded by plutus computational 
+limits. However, it is hard to make individual functions for each possible statement accepted. 
+
+## Plonk verification
+Plonk verification is quite complex, and might differ depending on the number of custom gates, or specific wire 
+connections. In this section we expose the verifier complexity as described 
+[here](https://hackmd.io/Vr0DUYPCR0iWADpO8kpkAA?view#Verifier-Algorithm). We count every challenge computation 
```

Comment:
Better use a reference to the [paper](https://eprint.iacr.org/2019/953.pdf) as it is more updated. 
In particular the current version has a new way of computing the linearisation polynomial that saves some operations for the verifier.